e3nn/o3/_tensor_product/_tensor_product.py

```diff
@@ -143,16 +143,14 @@ class TensorProduct(CodeGenMixin, torch.nn.Module):
     ...         if ir_out in ir_1 * ir_2
     ...     ]
     ... )
-    >>> ws = []
-    >>> for ins in module.instructions:
-    ...     if ins.has_weight:
-    ...         weight = torch.empty(ins.path_shape)
+    >>> with torch.no_grad():
+    ...     for weight in module.weight_views():
     ...         mul_1, mul_2, mul_out = weight.shape
     ...         # formula from torch.nn.init.xavier_uniform_
     ...         a = (6 / (mul_1 * mul_2 + mul_out))**0.5
-    ...         ws += [weight.uniform_(-a, a).view(-1)]
-    >>> with torch.no_grad():
-    ...     module.weight[:] = torch.cat(ws)
+    ...         new_weight = torch.empty_like(weight)
+    ...         new_weight.uniform_(-a, a)
+    ...         weight[:] = new_weight
```

Comment:
This line cause the problem in the tests, I don't know why but for any line that returns a non None value like it is for `new_weight.uniform_(-a, a)` pytest compare that returned value with the expected output.

```suggestion
    ...         weight[:].uniform_(-a, a)
```